### PR TITLE
Dispatch custom event when editor is ready

### PR DESCRIPTION
**Pull request by ivarnakken**
I believe this will be useful for our e2e spec, but I haven't tested it in practice because I can't get the local package linking to work ..

### CODE REVIEW

**Comment by eikhr** [comment]
I agree, it should be possible to actually wait for the editor to be ready with this:)

**Review by eikhr — APPROVED** [review]

File: src/index.tsx [hunk 1]
```diff
@@ -259,6 +259,11 @@ const LegoEditor = (props: Props): JSX.Element => {
       : initialValue,
   );
 
+  useEffect(() => {
+    const event = new CustomEvent('editorReady');
+    window.dispatchEvent(event);
+  }, []);
+
```
Comment (eikhr) [inline src/index.tsx:266]: Actually on second thought I'm not sure. I think maybe the Slate-component needs some time to initialize, and this wouldn't take that into account

**Review by norbye — APPROVED** [review]

File: src/index.tsx (hunk 1 above)
Comment (norbye) [reply src/index.tsx:266]: Not sure if it'll work 100% either, but we might as well test it out and see how it works :shipit: tried researching for other options but can't seem to find many better alternatives

**Review by eikhr — CHANGES_REQUESTED** [review]
Unfortunately, this doesn't work. I tested it a bit locally (by using `yarn pack` to get a `tar.gz` archive of the new version and then installing it using "file:\<filename>" in package.json of lego-webapp). Even when I was finally able to make cypress wait for the "editorReady"-event, it still fails about 1/3 of the times when the first action is typing into the editor.

**Comment by ivarnakken** [comment]
@eikhr I only get errors when I do exactly what you do for testing ...

However, do you think it would help if we added `value` as a dependency to the `useEffect`?

**Comment by eikhr** [comment]
I don't think so. The event emitter works fine and we get one event when the component is initially rendered, but it seems Slate is still not ready to accept input at this time. I think there is something internally in Slate that runs async and must be finished before it starts accepting inputs. I'm not really sure how we can detect that.

**Pull request closed without merging**